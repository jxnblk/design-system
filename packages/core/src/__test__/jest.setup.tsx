import React from 'react'
import { ThemeProvider } from 'styled-components'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import { createTheme } from '..'

global.rendererCreateWithTheme = (tree) =>
  renderer.create(<ThemeProvider theme={createTheme()}>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
