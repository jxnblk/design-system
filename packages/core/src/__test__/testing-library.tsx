import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
