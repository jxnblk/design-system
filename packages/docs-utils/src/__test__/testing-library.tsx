import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
