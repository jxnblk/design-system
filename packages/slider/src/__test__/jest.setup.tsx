import React from 'react'
import { ThemeProvider } from 'pcln-design-system'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

global.rendererCreateWithTheme = (tree) => renderer.create(<ThemeProvider>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
