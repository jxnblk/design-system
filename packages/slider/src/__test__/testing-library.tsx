import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
