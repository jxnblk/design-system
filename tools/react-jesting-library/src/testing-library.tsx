import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

/**
 * customRender wraps the render method of \@testing-library/react and wraps it in our
 * custom AllTheProviders which provides a pcln-design-system theme
 *
 * @public
 */
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
