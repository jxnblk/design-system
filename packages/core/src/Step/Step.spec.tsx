import React from 'react'
import { render, fireEvent } from '../__test__/testing-library'

import { Step } from '.'

describe('Step', () => {
  it('should render correctly with no props without throwing', () => {
    expect(() => {
      const { asFragment } = render(<Step>Step</Step>)

      expect(asFragment()).toMatchSnapshot()
    }).not.toThrow()
  })

  it('should render an active step correctly', () => {
    const { asFragment, container, getByTestId } = render(
      <Step active data-testid='Step'>
        Step
      </Step>
    )

    expect(asFragment()).toMatchSnapshot()

    expect(getByTestId('Step')).toHaveStyleRule('cursor', 'default')

    expect(container.querySelector('svg')).not.toBeInTheDocument()
  })

  it('should render a completed step correctly', () => {
    const { asFragment, container, getByTestId } = render(
      <Step completed data-testid='Step'>
        Step
      </Step>
    )

    expect(asFragment()).toMatchSnapshot()

    expect(getByTestId('Step')).toHaveStyleRule('cursor', 'default')

    expect(container.querySelector('svg')).toBeInTheDocument()
  })

  it('should render a step that is "active" and "completed" correctly', () => {
    const { asFragment, container, getByTestId } = render(
      <Step active completed data-testid='Step'>
        Step
      </Step>
    )

    expect(asFragment()).toMatchSnapshot()

    expect(getByTestId('Step')).toHaveStyleRule('cursor', 'default')

    expect(container.querySelector('svg')).toBeInTheDocument()
  })

  it('should handle onClick', () => {
    const handleClick = jest.fn()

    const { asFragment, getByTestId, getByText } = render(
      <Step active data-testid='Step' onClick={handleClick}>
        Step
      </Step>
    )

    expect(asFragment()).toMatchSnapshot()

    expect(getByTestId('Step')).toHaveStyleRule('cursor', 'pointer')

    fireEvent.click(getByText('Step'))
    expect(handleClick).toHaveBeenCalled()
  })
})
