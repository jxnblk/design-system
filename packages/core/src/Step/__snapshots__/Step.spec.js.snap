// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should handle onClick 1`] = `
<DocumentFragment>
  .c2 {
  color: #007aff;
  font-weight: 700;
  font-size: 14px;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: pointer;
  padding: 0;
  font-weight: unset;
}

.c1:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c1:hover {
  background-color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <button
      class="c1"
      data-testid="Step"
    >
      <div
        aria-selected="true"
        class="c2"
        color="primary"
        font-size="1"
      >
        Step
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Step should render a completed step correctly 1`] = `
<DocumentFragment>
  .c4 {
  color: #007aff;
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 4px;
  width: 16px;
  color: #007aff;
}

.c3 {
  outline: none;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: default;
  padding: 0;
  font-weight: unset;
}

.c1:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c1:hover {
  background-color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <button
      class="c1"
      data-testid="Step"
    >
      <svg
        aria-hidden="false"
        aria-label="Step Completed"
        class="c2 c3"
        color="primary"
        fill="currentcolor"
        focusable="false"
        height="16"
        role="img"
        tabindex="-1"
        viewBox="0 0 24 24"
        width="16"
      >
        <path
          d="M8.6 15.6l-4.2-4.2L3 12.8l5.6 5.6 12-12L19.2 5 8.6 15.6z"
        />
      </svg>
      <div
        aria-selected="false"
        class="c4"
        color="primary"
        font-size="1"
      >
        Step
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Step should render a step that is "active" and "completed" correctly 1`] = `
<DocumentFragment>
  .c4 {
  color: #007aff;
  font-weight: 700;
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 4px;
  width: 16px;
  color: #007aff;
}

.c3 {
  outline: none;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: default;
  padding: 0;
  font-weight: unset;
}

.c1:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c1:hover {
  background-color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <button
      class="c1"
      data-testid="Step"
    >
      <svg
        aria-hidden="false"
        aria-label="Step Completed"
        class="c2 c3"
        color="primary"
        fill="currentcolor"
        focusable="false"
        height="16"
        role="img"
        tabindex="-1"
        viewBox="0 0 24 24"
        width="16"
      >
        <path
          d="M8.6 15.6l-4.2-4.2L3 12.8l5.6 5.6 12-12L19.2 5 8.6 15.6z"
        />
      </svg>
      <div
        aria-selected="true"
        class="c4"
        color="primary"
        font-size="1"
      >
        Step
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Step should render an active step correctly 1`] = `
<DocumentFragment>
  .c2 {
  color: #007aff;
  font-weight: 700;
  font-size: 14px;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: default;
  padding: 0;
  font-weight: unset;
}

.c1:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c1:hover {
  background-color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <button
      class="c1"
      data-testid="Step"
    >
      <div
        aria-selected="true"
        class="c2"
        color="primary"
        font-size="1"
      >
        Step
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Step should render correctly with no props without throwing 1`] = `
<DocumentFragment>
  .c2 {
  color: #4f6f8f;
  font-size: 14px;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: default;
  padding: 0;
  font-weight: unset;
}

.c1:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c1:hover {
  background-color: #fff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        aria-selected="false"
        class="c2"
        color="text.light"
        font-size="1"
      >
        Step
      </div>
    </button>
  </div>
</DocumentFragment>
`;
